openedx/adg/lms/applications/views.py

```diff
@@ -263,3 +266,100 @@ def get_context_data(self, request):
                     'birth_year': extended_profile.birth_date.year,
                 })
         return context
+
+
+class CoverLetterView(RedirectToLoginOrRelevantPageMixin, View):
+    """
+    View enabling the user to select a Business Line and upload or write a cover letter.
+    """
+
+    template_name = 'adg/lms/applications/cover_letter.html'
+    login_url = '/register'
+
+    def is_precondition_satisfied(self):
+        """
+        Checks if a written application is already submitted or not.
+        Returns:
+            bool: True if written application is not completed, False otherwise.
+        """
+        user_application_hub, _ = ApplicationHub.objects.get_or_create(user=self.request.user)
+
+        return not user_application_hub.is_written_application_completed
+
+    def handle_no_permission(self):
+        """
+        Redirects to application hub on get request or returns http 400 on post request.
+        Returns:
+            HttpResponse object.
+        """
+        if self.request.method == 'POST':
+            return HttpResponse(status=400)
+        else:
+            return redirect('application_hub')
+
+    def get(self, request):
+        """
+        Send the context data for example a list of business lines and saved user_application.
+        Returns:
+            HttpResponse object.
+        """
+        business_lines = BusinessLine.objects.all()
+        file_name = None
+
+        try:
+            registration_business_line = request.user.extended_profile.company
+        except ExtendedUserProfile.DoesNotExist:
+            registration_business_line = None
+
+        try:
+            user_application = request.user.application
+            if user_application.cover_letter_file:
+                file_name = Path(user_application.cover_letter_file.name).name
```

Comment:
file name is a generic name, let's please rename it to some specific name. Also I guess it is the complete path not just the file name